moderntimeline.dtx

```diff
@@ -487,6 +558,10 @@
    \def\tl@firstyear{#1}
    \def\tl@lastyear{#2}
    \pgfmathsetmacro\tl@yearrange{\tl@lastyear-\tl@firstyear}
+   \pgfmathsetmacro\tl@totalyears{\tl@lastyear-\tl@firstyear}%
+   \pgfmathsetmacro\tl@totalmonths{12*\tl@totalyears}%
+   \pgfmathsetmacro\tl@tickdistmo{\hintscolumnwidth/\tl@totalmonths}%
+   \pgfmathsetmacro\tl@tickdistyr{\hintscolumnwidth/\tl@totalyears}%
```

Comment:
These two lines need to be computed outside of the macro definition, in order to make it optional.